zip/src/main/kotlin/io/github/paulgriffith/kindling/zip/ZipView.kt

```diff
@@ -269,22 +203,30 @@ object ZipViewer : Tool {
         put(ProjectView::isProjectDirectory, ::ProjectView)
     }
 
-    fun createView(fileSystemProvider: FileSystemProvider, vararg paths: Path): Result<PathView> = runCatching {
+    fun createView(fileSystemProvider: FileSystemProvider, vararg paths: Path): PathView? = runCatching {
         if (paths.size > 1) {
             MultiToolView(fileSystemProvider, paths.toList())
         } else {
             val path = paths.first()
-            handlers.firstNotNullOfOrNull { (predicate, provider) ->
-                if (predicate(path)) {
-                    provider(fileSystemProvider, path)
-                } else {
-                    null
+            val specificView = handlers.firstNotNullOfOrNull { (predicate, provider) ->
+                try {
+                    if (predicate(path)) {
+                        return@firstNotNullOfOrNull provider(fileSystemProvider, path)
+                    }
+                } catch (_: ToolOpeningException) {
                 }
-            } ?: GenericFileView(fileSystemProvider, path)
+                null
+            }
+            specificView ?: if (path.isRegularFile()) {
+                GenericFileView(fileSystemProvider, path)
+            } else null
         }
+    }.getOrElse { ex ->
+        logger.error("Failed to open ${paths.contentToString()}", ex)
+        null
     }
```

Comment:
This got gross; my simplification was losing the semantics of "try a tool to open; if the tool fails try a generic file view" and would just bomb out as soon as it failed to open a tool view